### PR TITLE
Check if `tmp` is pointer prior to create load

### LINKED ISSUE #1443: specfun.f: AssertFailed: t->isPointerTy()
```console
$ lfortran --show-llvm --fixed-form --implicit-typing --implicit-interface ./scipy/special/specfun/specfun.f
Internal Compiler Error: Unhandled exception
Traceback (most recent call last):
  File "/Users/ondrej/repos/lfortran/lfortran/src/bin/lfortran.cpp", line 1870
    return emit_llvm(arg_file, lfortran_pass_manager,
  File "/Users/ondrej/repos/lfortran/lfortran/src/bin/lfortran.cpp", line 777
    = fe.get_llvm(input, lm, pass_manager, diagnostics);
  File "/Users/ondrej/repos/lfortran/lfortran/src/lfortran/fortran_evaluator.cpp", line 287
    Result<std::unique_ptr<LLVMModule>> res = get_llvm2(code, lm, pass_manager, diagnostics);
  File "/Users/ondrej/repos/lfortran/lfortran/src/lfortran/fortran_evaluator.cpp", line 309
    diagnostics, lm.files.back().in_filename);
  File "/Users/ondrej/repos/lfortran/lfortran/src/lfortran/fortran_evaluator.cpp", line 342
    compiler_options, run_fn, infile);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 7945
    v.nested_func_types = pass_find_nested_vars(asr, context,
  File "../libasr/asr.h", line 4547
  File "../libasr/asr.h", line 4523
  File "../libasr/asr.h", line 4548
  File "../libasr/asr.h", line 4277
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 1343
    visit_symbol(*item.second);
  File "../libasr/asr.h", line 4550
  File "../libasr/asr.h", line 4287
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 3794
    visit_procedures(x);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4155
    this->visit_stmt(*x.m_body[i]);
  File "../libasr/asr.h", line 4565
  File "../libasr/asr.h", line 4321
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5282
    }, [=]() {
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 331
    start_new_block(elseBB); {
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5284
    this->visit_stmt(*x.m_orelse[i]);
  File "../libasr/asr.h", line 4565
  File "../libasr/asr.h", line 4321
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5282
    }, [=]() {
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 331
    start_new_block(elseBB); {
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5284
    this->visit_stmt(*x.m_orelse[i]);
  File "../libasr/asr.h", line 4565
  File "../libasr/asr.h", line 4308
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4729
    this->visit_expr_wrapper(x.m_value, true);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4921
    this->visit_expr(*x);
  File "../libasr/asr.h", line 4611
  File "../libasr/asr.h", line 4426
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 6275
    this->visit_expr_wrapper(x.m_arg, true);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4921
    this->visit_expr(*x);
  File "../libasr/asr.h", line 4611
  File "../libasr/asr.h", line 4376
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 5625
    this->visit_expr_wrapper(x.m_left, true);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 4925
    if( load_ref ) {
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/asr_to_llvm.cpp", line 286
    return LLVM::CreateLoad(*builder, x);
  File "/Users/ondrej/repos/lfortran/lfortran/src/libasr/codegen/llvm_utils.cpp", line 12
    LCOMPILERS_ASSERT(t->isPointerTy());
AssertFailed: t->isPointerTy()
```

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #1443.

### CODE REVIEW

**Review by certik — APPROVED** [review]
It looks good to me, but I want @czgdp1807 to review also, to make sure this is consistent with the rest of the LLVM backend.

**Review by czgdp1807 — APPROVED** [review]
See https://github.com/lfortran/lfortran/pull/1451/commits/4c5710f9b58b8dd8ce04c3697e368206cd266c6c#r1143609406 for explanation.

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 1]
```diff
@@ -4950,7 +4950,10 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             x->type == ASR::exprType::ArraySection ||
             x->type == ASR::exprType::StructInstanceMember ) {
             if( load_ref ) {
-                tmp = CreateLoad(tmp);
+                // check if the tmp is a pointer
+                if( tmp->getType()->isPointerTy() ) {
+                    tmp = CreateLoad(tmp);
+                }
```
[the remark this answers is not in the document]
Comment (Pranavchiku) [reply src/libasr/codegen/asr_to_llvm.cpp]: Okay, yes! Thanks for improving this!

**Comment by certik** [comment]
Perfect, thanks!

**Pull request merged**